src/tests/EShopworld.WorkerProcess.IntegrationTests/WorkerLeaseTests.cs

```diff
@@ -178,15 +217,15 @@ private void SetupWorkerLeases(int leaseCount, Func<int,int> assignPriorityFunc)
                 var telemetry = _serviceProvider.GetService<IBigBrother>();
                 var leaseStore = _serviceProvider.GetService<ILeaseStore>();
                 var slottedInterval = _serviceProvider.GetService<ISlottedInterval>();
-                var timer = _serviceProvider.GetService<ITimer>();
+                var timer = new SystemTimer();
```

Comment:
I am curious about what difference this makes.

Reply:
it does, as currently the timer is created as a singleton, while we want each process using its own timer. Makes sense?